chainer/links/model/classifier.py

```diff
@@ -53,6 +54,8 @@ def __call__(self, x, t):
         self.accuracy = None
         self.y = self.predictor(x)
         self.loss = self.lossfun(self.y, t)
+        reporter.report({'loss': self.loss}, self)
```

Comment:
The report function does nothing if no reporter is set as current, thus the backward compatibility is not broken.